direct/core/src/main/java/cz/o2/proxima/direct/core/DirectDataOperator.java

```diff
@@ -66,13 +69,14 @@
   private static final String THROUGHPUT_LIMITER_PREFIX = "direct.throughput-limiter.";
   private static final String KW_CLASS = "class";
   private static final AtomicInteger threadId = new AtomicInteger();
+  private static final String ID = UUID.randomUUID().toString();
 
   private static Factory<ExecutorService> createExecutorFactory() {
     return () ->
         Executors.newCachedThreadPool(
             r -> {
               Thread t = new Thread(r);
-              t.setName(String.format("DirectDataOperatorThread-%d", threadId.incrementAndGet()));
+              t.setName(String.format("DirectDataOperatorThread-%s-%d", ID, threadId.incrementAndGet()));
```

Comment:
what is purpose of this? the ID is static, so it will remain the same for all instances within JVM

Reply:
It is the same for all instances of the same classloader.